### ISSUE TITLE
[experimental] OpenReward Standard environment adapter

### ISSUE THREAD

**Issue by adithya-s-k** [huggingface/trl#5695]
### Feature request

Add `trl.experimental.openreward.OpenRewardEnv` — a thin adapter that lets any TRL trainer with an `environment_factory` slot (GRPO, AsyncGRPO, RLOO, …) train against any environment that speaks the [**Open Reward Standard (ORS)**](https://openrewardstandard.io) — the open HTTP/SSE protocol for RL environments — whether the environment is hosted on the [**OpenReward platform**](https://openreward.ai), self-hosted on your own infrastructure, or running locally for development. One string identifier is enough.

One class, three properties wired into TRL's three trainer slots:

```python
from trl import GRPOConfig, GRPOTrainer
from trl.experimental.openreward import OpenRewardEnv

env = OpenRewardEnv("Eigent/SETA")  # or "http://localhost:8000"

trainer = GRPOTrainer(
    model="Qwen/Qwen3-4B",
    args=GRPOConfig(...),
    train_dataset=env.dataset,           # auto-derived from the env's task endpoints
    environment_factory=env.factory,     # per-rollout env with bound tools
    reward_funcs=env.reward_func,        # ORS outcome reward
)
```

Implementation wraps the official `openreward` SDK (so protocol details are tracked upstream) and dynamically binds one Python method per ORS tool with a typed signature + docstring, so `inspect.getmembers` and `transformers.utils.get_json_schema` work with no per-env wrapper code.

Optional dep: `pip install trl[openreward]`. Lazy import — non-users pay nothing. Pattern matches `trl[peft]`, `trl[vllm]`.

### Motivation

[**OpenReward**](https://openreward.ai) is an open ecosystem for RL environments centered on the [**Open Reward Standard (ORS)**](https://openrewardstandard.io) — a public, language-agnostic specification (HTTP/REST + SSE) for how an RL environment exposes itself: a uniform set of endpoints for tasks/splits, session lifecycle, tool discovery and tool calling, with tools described as JSON Schema and a reward + `finished` signal returned by the environment's own grader. (Per [the standard](https://openrewardstandard.io), ORS and MCP are *complementary* — MCP for general tool access, ORS adds RL primitives: episodes, reward signals, task splits, curriculum management.) Because ORS is *just a standard*, an environment is portable by construction:

- **Hosted on the OpenReward platform** (`openreward.ai`) — fully managed sandbox + bucket, e.g. `Eigent/SETA`, `kanishk/EndlessTerminals`, `nebius/SWE-rebench-V2`.
- **Self-hosted on any container-running service** — your own infra, a Docker host, K8s, anywhere an HTTP server can run. The environment is a normal process you operate.
- **Run locally for development** — the same env runs on `localhost:8000` as a regular FastAPI server, so authoring a new environment looks like authoring a normal web service: write the env, hit it with `curl`, iterate. When it's ready, deploy the same image to the OpenReward platform or any container host without changing the contract.

This portability is what makes the ecosystem interesting for TRL users: an env author can develop against `localhost`, an org can self-host sensitive envs behind their own VPC, and a researcher can pick any public env on `openreward.ai` — all behind the same wire protocol.

Today, plugging an ORS env into TRL means re-implementing the same wire per env: HTTP client, tool-schema → Python signature, session lifecycle, reward extraction, etc. Each env maintainer ends up shipping a one-off integration script. A single adapter at the protocol level eliminates that duplication and lets every TRL trainer talk to every ORS env (current or future) by string identifier alone.

Symmetric to the OpenEnv path (`OpenEnv` ships one Python package per env with bespoke `Action`/`Observation` types → loose, examples-only integration is a good fit). ORS is the opposite: protocol uniformity → a single shared adapter pays back across every env.

Refs:
- OpenReward platform: https://openreward.ai
- Open Reward Standard: https://openrewardstandard.io
- OpenReward docs: https://docs.openreward.ai
- `openreward` Python SDK: https://pypi.org/project/openreward/
- openreward-cookbook (SLIME / Tinker / Miles / SkyRL integration patterns we cross-referenced): https://github.com/OpenRewardAI/openreward-cookbook

**Comment by adithya-s-k** [comment]
PR is open: #5696. After review feedback the public surface is now:

```python
spec = OpenRewardSpec("Eigent/SETA")  # or "http://localhost:8000"

trainer = GRPOTrainer(
    train_dataset=spec.train_dataset,
    environment_factory=spec.environment_factory,
    reward_funcs=spec.reward_funcs,
    ...
)
```

Each spec property maps 1:1 to a trainer kwarg — the PR doesn't yet auto-derive `train_dataset` / `reward_funcs` from the env, but @qgallouedec raised that as a natural follow-up (separate issue, post-merge).

Class name landed as `OpenRewardSpec` rather than `OpenRewardEnv` since what you construct is really a spec/factory, not an env you step/reset.

Docs page + tests are next on the PR.
